tests/framework/db/mssql/QueryBuilderTest.php

```diff
@@ -68,36 +68,178 @@ public function testOffset()
+            'value' => $columnComment2,
+        ], $result[0]);
+    }
 
-        $expected = "sp_dropextendedproperty @name = N'MS_Description', @level1type = N'Table',  @level1name = comment";
-        $sql = $qb->dropCommentFromTable('comment');
-        $this->assertEquals($expected, $sql);
+    public function testCommentAdditionOnQuotedTableOrColumn()
+    {
+        $table = 'stranger \'table';
+        $tableComment = 'A comment for stranger \'table.';
+        $this->runAddCommentOnTable($tableComment, $table);
+        $resultTable = $this->getCommmentsFromTable($table);
+        $this->assertEquals([
+            'objtype' => 'TABLE',
+            'objname' => $table,
+            'name' => 'MS_description',
+            'value' => $tableComment,
+        ], $resultTable[0]);
+
+        $column = 'stranger \'field';
+        $columnComment = 'A comment for stranger \'field column in stranger \'table.';
+        $this->runAddCommentOnColumn($columnComment, $table, $column);
+        $resultColumn = $this->getCommentsFromColumn($table, $column);
+        $this->assertEquals([
+            'objtype' => 'COLUMN',
+            'objname' => $column,
+            'name' => 'MS_description',
+            'value' => $columnComment,
+        ], $resultColumn[0]);
+    }
+
+    public function testCommentRemovalFromTableAndFromColumn()
+    {
+        $table = 'profile';
+        $tableComment = 'A comment for profile table.';
+        $this->runAddCommentOnTable($tableComment, $table);
+        $this->runDropCommentFromTable($table);
+        $result = $this->getCommmentsFromTable($table);
+        $this->assertEquals([], $result);
+
+        $column = 'description';
+        $columnComment = 'A comment for description column in profile table.';
+        $this->runAddCommentOnColumn($columnComment, $table, $column);
+        $this->runDropCommentFromColumn($table, $column);
+        $result = $this->getCommentsFromColumn($table, $column);
+        $this->assertEquals([], $result);
+    }
+
+    public function testCommentRemovalFromQuotedTableOrColumn()
+    {
+        $table = 'stranger \'table';
+        $tableComment = 'A comment for stranger \'table.';
+        $this->runAddCommentOnTable($tableComment, $table);
+        $this->runDropCommentFromTable($table);
+        $result = $this->getCommmentsFromTable($table);
+        $this->assertEquals([], $result);
+
+        $column = 'stranger \'field';
+        $columnComment = 'A comment for stranger \'field in stranger \'table.';
+        $this->runAddCommentOnColumn($columnComment, $table, $column);
+        $this->runDropCommentFromColumn($table, $column);
+        $result = $this->getCommentsFromColumn($table, $column);
+        $this->assertEquals([], $result);
+    }
+
+    public function testCommentColumn()
+    {
+        $this->markTestSkipped("Testing the behavior, not sql generation anymore.");
+    }
+
+    public function testCommentTable()
```

Comment:
Can be removed, I think. @samdark ?

Reply:
Then parent test from `yiiunit\framework\db\QueryBuilderTest` execution won't be prevented. Isn't that the intention of having this method?

Reply:
Right, yes.